opentelemetry-api/src/opentelemetry/context/propagation/binaryformat.py

```diff
@@ -0,0 +1,58 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import abc
+import typing
+
+from opentelemetry.trace import SpanContext
+
+
+class BinaryFormat(abc.ABC):
```

Comment:
Do we want this to be an ABC? What about making it usable as a no-op propagator?

Reply:
There was a light discussion on that choice here:

https://github.com/open-telemetry/opentelemetry-python/pull/61#discussion_r307940312
https://github.com/open-telemetry/opentelemetry-python/pull/68#discussion_r309037931

I think there is value in implementing the base class as an ABC, as it provides fast feedback if the implementation is incomplete. I would vote we make a no-op propagator separately, maybe include it in API to not require the SDK.